dod/src/main/java/com/revolut/rxdata/dod/DataObservableDelegate.kt

```diff
@@ -105,7 +107,10 @@ class DataObservableDelegate<Params : Any, Domain : Any> constructor(
             val memCache = fromMemory(params)
             val memoryIsEmpty = memCache == null
             val subject = subject(params)
-            val loading = loadingStrategy.refreshMemory || memoryIsEmpty || failedNetworkRequests.containsKey(params)
+            val loading = loadingStrategy.refreshMemory
+                    || memoryIsEmpty
+                    || failedNetworkRequests.containsKey(params)
+                    || (needToBeRefreshed[params] == true && loadingStrategy.refreshStorage)
```

Comment:
Consider renaming _needToBeRefreshed_ -> _memoryCacheIsFromStorage_. 
memoryCacheIsFromStorage would reflect state explicitly without coupling to any further logic about its use. 
_needToBeRefreshed_ would also be confusing, as there're many points of refresh in DoD, and this map only represents a thin subset. 
The logic itself seems to become more comprehensive with this naming IMO: (I'd also group memory related checks nearby, unless order is of some importance, which it shouldn't be)

```kotlin

  private val memoryCacheIsFromStorage = ConcurrentHashMap<Params, Boolean>()

    val loading = loadingStrategy.refreshMemory
                        || memoryIsEmpty
                        || (memoryCacheIsFromStorage[params] == true && loadingStrategy.refreshStorage)
                        || failedNetworkRequests.containsKey(params)

```